### PR TITLE
Fixes #20247 - Add welcome page to foreman discovery

### PR DESCRIPTION
**Pull request by rabajaj0509**

### CODE REVIEW

File: app/controllers/discovered_hosts_controller.rb [hunk 1]
```diff
@@ -16,6 +16,13 @@ class DiscoveredHostsController < ::ApplicationController
 
   layout 'layouts/application'
 
+  def welcome
+    if (Host::Discovered.first.blank?)
+      @welcome = true
+      render :welcome
+    end
+  end
+
```
Comment (lzap) [inline app/controllers/discovered_hosts_controller.rb]: Can you try to override `model_of_controller` instead? Will it work?

File: app/views/discovered_hosts/welcome.html.erb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<% content_for(:title, _("DiscoveredHosts")) %>
```
Comment (lzap) [inline app/views/discovered_hosts/welcome.html.erb]: Why not `_("Discovered Hosts")` it's a visible title, isn't it?

File: app/views/discovered_hosts/welcome.html.erb [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+<div class="blank-slate-pf">
+  <div class="blank-slate-pf-icon">
+    <%= icon_text("gears", "", :kind => "fa") %>
+  </div>
+  <h1><%= _('Discovered Hosts') %></h1>
+  <p><%= _("You don't have any visible discovered hosts. This plugin enables Foreman to do automatic bare-metal discovery of unknown nodes on the provisioning network.") %></p>
```
Comment (lzap) [inline app/views/discovered_hosts/welcome.html.erb]: Can you reword and split it to i18n keys:

    <p><%= _("No discovered hosts found in this context.") %>
    <%= _("This page shows discovered bare-metal or virtual nodes waiting to be provisioned.") %></p>

File: app/views/discovered_hosts/welcome.html.erb [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+  <p><%= link_to _('Learn more about this in the documentation.'), documentation_url("8.0/index.html", {:root_url => 'https://theforeman.org/plugins/foreman_discovery/'})%></p>
```
Comment (lzap) [inline app/views/discovered_hosts/welcome.html.erb]: Use something like:

    ForemanDiscovery::VERSION.scan(/\d+\.\d+/).first

to get the current stable version number for the URL. Docs are just about to appear there this week.

**Review by lzap — CHANGES_REQUESTED** [review]
Good, few things.

**Comment by lzap** [comment]
[test]

**Comment by lzap** [comment]
Particular reason why you haven't reworded the text? I *think* this would be a good fit:

```
<p><%= _("No discovered hosts found in this context.") %>
<%= _("This page shows discovered bare-metal or virtual nodes waiting to be provisioned.") %></p>
```

**Comment by rabajaj0509** [comment]
@lzap , (sorry), done :)

**Comment by lzap** [comment]
[test]

**Comment by lzap** [comment]
Thanks a lot.

**Pull request merged**